app/controllers/redirect/redirect_controller.rb

```diff
@@ -10,14 +10,35 @@ def location_params
       params.permit(*PERMITTED_LOCATION_PARAMS).to_h.symbolize_keys
     end
 
+    def partner_query_params
+      @partner_query_params ||= begin
+        {
+          partner: current_sp&.agency&.name,
+          partner_div: current_sp&.integration&.name,
+        }.compact
+      end
```

Comment:
is this method called multiple times? it seems cheap to call (`current_sp` will cache its DB call) so I don't think it needs memoization
```suggestion
      {
        partner: current_sp&.agency&.name,
        partner_div: current_sp&.integration&.name,
      }.compact
```